### PR TITLE
repo: add new README

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
Download failure during build.

File: src/lxc/conf.c [hunk 1]
```diff
@@ -3384,27 +3384,32 @@ int lxc_assign_network(const char *lxcpath, char *lxcname,
...
+	if (ret < 0 || ret >= MAXPATHLEN) {
+		ERROR("failed to create path \"%s\"", path);
 		return -E2BIG;
 	}
-	f = fopen(path, "w");
-	if (!f) {
-		perror("open");
-		return -EINVAL;
+
+	fd = open(path, O_RDWR);
```
Comment (Blub) [inline src/lxc/conf.c]: You went from `w` to `O_RDWR` - shouldn't `O_WRONLY` suffice? I see no `read` calls.

Comment (brauner) [inline src/lxc/conf.c]: Yeah, good call!

File: README.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+LXC also supports at least the following C standard libraries:
+
+- glibc
+- musl
+- bionic (Android's libc)
+
+## Backwards Compatibility
+
+LXC has always focused on strong backwards compatiblity. In fact, the API
```
Comment (hallyn) [inline README.md]: compatibility mis-spelled.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+introduction at:
+
+- https://www.kernel.org/doc/html/v4.10/process/coding-style.html
+
+and should also take a look at the [CONTRIBUTING](CONTRIBUTING) file in this
+repo.
+
+If you want to become more active it is usually also a good idea to show up in
+the LXC IRC channel `#lxc-dev` on `Freenode`. We try to do all development out
+in the open and discussion of new features or bugs is done either in approriate
```
Comment (hallyn) [inline README.md]: appropriate mis-spelled

File: README.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+Github issues or on IRC.
+
+When thinking about making security critical contributions or substantial
+changes it is usally a good idea to ping the developers first and ask whether
```
Comment (hallyn) [inline README.md]: usally

File: README.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+We maintain an discuss forum at
+
+- https://discuss.linuxcontainers.org/
+
+where you can get support.
+
+### IRC
+
+You can find support by joining `#lxccontainers` on `Freenode`.
```
Comment (hallyn) [inline README.md]: it's "lxcontainers" right?

Comment (brauner) [inline README.md]: oh you're right

**Comment by brauner** [comment]
@hallyn, updated.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+   devices for an unprivileged user (see LXC's `lxc-user-nic` binary) the only
+   option is to share the network namespace with the host. Although this should
+   be secure in principle, sharing the host's network namespace is still one
+   step of isolation less and increases the attack vector.
+   Furthermore, when host and container share the same network namespace the
+   kernel will refuse any sysfs mounts. This usually means that the init binary
+   inside of the container will not be able to boot up correctly.
+
+2. User Namespaces: As outlined above, user namespaces are a big security
+   enhancement. However, users which are unprivileged on the host will only be
```
Comment (hallyn) [inline README.md]: Oh, sorry - this sentence is confusing.  Can you turn that around, something like

"However, without relying on privileged helpers users who are unprivileged on the host are only permitted to map their own UID into a container."

Comment (brauner) [inline README.md]: Yup, one sec.

Comment (brauner) [inline README.md]: @hallyn, see https://github.com/lxc/lxc/pull/1594 .

**Pull request merged**